File: app/src/main/java/fr/frogdevelopment/nihongo/dico/SplashScreenActivity.java
package fr.frogdevelopment.nihongo.dico;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import fr.frogdevelopment.nihongo.dico.to_delete.MainActivity;

public class SplashScreenActivity extends Activity {

    // Splash screen timer
    private static final long SPLASH_TIME_OUT = 500;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_screen_activity);

        // This method will be executed once the timer is over
        new Handler().postDelayed(this::launchActivity, SPLASH_TIME_OUT);
    }

    private void launchActivity() {
        startActivity(new Intent(this, MainActivity.class));

        // close this activity
        finish();
    }

}

File: app/src/main/java/com/frogdevelopment/nihongo/dico/ui/SplashScreenActivity.kt
package com.frogdevelopment.nihongo.dico.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import com.frogdevelopment.nihongo.dico.R

class SplashScreenActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_screen_activity)
        // This method will be executed once the timer is over
        Handler(Looper.myLooper()!!).postDelayed({ launchActivity() }, SPLASH_TIME_OUT)
    }

    private fun launchActivity() {
        startActivity(Intent(this, MainActivity::class.java))
        // close this activity
        finish()
    }

    companion object {
        // Splash screen timer
        private const val SPLASH_TIME_OUT: Long = 500
    }
}